pkg/cableengine/ipsec/driver.go

```diff
@@ -0,0 +1,22 @@
+package ipsec
+
+import (
+	"github.com/submariner-io/submariner/pkg/types"
+)
+
+// Driver is used by the ipsec engine to actually connect the tunnels
+type Driver interface {
+
+	// Initialize the driver with any state it needs
+	Init() error
+
+	// Returns an array of all the active connections for the given cluster
+	GetActiveConnections(clusterID string) ([]string, error)
+
+	// Connects to the given endpoint and reutrns a string representation of
+	// the IP address of the target endpoint
+	ConnectToEndpoint(endpoint types.SubmarinerEndpoint) (string, error)
+
+	// Disconnects from the endpoint identified by the cable ID
```

Comment:
`// DisconnectFromEndpoint disconnects from the endpoint identified by the cable ID.`

On a side note, the one thing that's a bit esoteric is the meaning and relationship of _cableID_ with respect to the `SubmarinerEndpoint` passed to `ConnectToEndpoint`. _cableID_  is actually passed upstream from `SubmarinerEndpoint.Spec.CableName` so it is assumed that the `CableEngine`  implementation uses `SubmarinerEndpoint.Spec.CableName` as an internal identifier in `ConnectToEndpoint`.  To alleviate this undocumented "tribal knowledge", I think we should pass `endpoint types.SubmarinerEndpoint` to this method as well and let the implementation decide how to identify a connection internally.

Reply:
I agree with you, but since this is a "change as little as possible" refactor I suggest we can do that in another patch since we also need to change the caller of RemoveCable itself (or change it to fetch the cable).

As for the comment I fixed that.

Reply:
@tpantelis make a very good point, but I agree with @mkolesnik about doing it on a separate patch.

Reply:
I agree as well - I didn't mean to do it in this patch - I should've made that clear.